File: Frontend/src/app/service/property.service.ts
import { Injectable } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs";
import { Property } from "../model/property";

@Injectable({
  providedIn: "root",
})
export class PropertyService {
  apiUrl = "";
  constructor(private http: HttpClient) {}

  getAllProperties() {
    //write your logic here
  }

  getPropertyById(id: number) {
    //write your logic here
  }

  createProperty(property: Property) {
    //write your logic here
  }

  updateProperty(property: Property) {
    //write your logic here
  }

  deleteProperty(id: number) {
    //write your logic here
  }

  searchProperties(queryParams: any) {
    //write your logic here
  }

  searchPropertiesByName(name: String): any {
    //write your logic here
  }

  searchPropertiesByPrice(price: Number): any {
    //write your logic here
  }

  searchPropertiesByCategory(category: String): any {
    //write your logic here
  }
}

File: Backend/src/main/java/com/example/propertylist/service/PropertyService.java
package com.example.propertylist.service;

import java.util.List;

import com.example.propertylist.dto.PropertyDTO;
import com.example.propertylist.entity.PropertyCategory;

public interface PropertyService {
	
	PropertyDTO createProperty(PropertyDTO propertyDTO);

	PropertyDTO getPropertyById(Long id);

	List<PropertyDTO> getAllProperties();

	PropertyDTO updateProperty(Long id, PropertyDTO propertyDTO);

	boolean deleteProperty(Long id);

	List<PropertyDTO> searchPropertiesByName(String name);

	List<PropertyDTO> searchPropertiesByPrice(Double price);

	List<PropertyDTO> searchPropertiesByCategory(PropertyCategory category);
}
